openfe/protocols/openmm_rfe/equil_rfe_methods.py

```diff
@@ -243,6 +247,8 @@ def _validate_alchemical_components(
                 logger.warning(wmsg)
                 warnings.warn(wmsg)  # TODO: remove this once logging is fixed
 
+    return mapping[0]  # type: ignore
```

Comment:
The reason why this by default was not returning mapping[0] was so that a future protocol could re-use this method without enforcing the same behaviour. I.e. a requirement that only a single mapping is necessary is specific to this specific Protocol. If we're enforcing that specific protocols' behaviour then maybe we should just make thsi a class method?

Reply:
https://github.com/OpenFreeEnergy/openfe/pull/666/files#diff-1ab9cb2879dcf6337e58cc145548646e494814de8535f1c23b56c13ee9162632R209-R213

currently this function only allows a single mapping, so the function returns the only allowable input.

I'm not sure this code is sufficiently complex/interesting that it can't just be rewritten for new protocols, it's just a bunch of checks specific to this Protocol.  Trying to make it general/useful for many different cases is probably more effort than just rewriting these checks.  The choice of function/staticmethod is not that different?